SPECS/mariner-rpm-macros/macros

```diff
@@ -102,6 +102,27 @@
   LDFLAGS="${LDFLAGS:-%{build_ldflags}}" ; export LDFLAGS ; \
   LT_SYS_LIBRARY_PATH="${LT_SYS_LIBRARY_PATH:-%_libdir:}" ; export LT_SYS_LIBRARY_PATH
 
+# Internal-only.  Do not use.  Expand a variable and strip the flags
+# not suitable to extension builders.
+%__extension_strip_flags() %{lua:
+local name = rpm.expand("%{1}")
+local value = " " .. rpm.expand("%{build_" .. name .. "}")
```

Comment:
Since we're appending `build_` I'm guessing we're expecting to be operating always on `build_X` macros. If I understood this correctly, what do you think about passing a full macro name like `%{__extension_strip_flags build_cflags}` to avoid confusion on what this function/macro operates on.

Reply:
Good idea! I went ahead and implemented it as you specified.